core/src/main/scala/cats/syntax/functor.scala

```diff
@@ -1,4 +1,28 @@
 package cats
 package syntax
 
-trait FunctorSyntax extends Functor.ToFunctorOps
+import cats.evidence.<~<
+
+trait FunctorSyntax extends Functor.ToFunctorOps {
+  implicit final def catsSyntaxFunctorOps[F[_], A](fa: F[A]): FunctorOps[F, A] =
+    new FunctorOps[F, A](fa)
+}
+
+final class FunctorOps[F[_], A](private val fa: F[A]) extends AnyVal {
+
+  /**
+   * Un-zips an `F[(A, B)]` consisting of element pairs or Tuple2 into two separate F's tupled.
+   *
+   * NOTE: Check for effect duplication, possibly memoize before
+   *
+   * {{{
+   * scala> import cats.Id
+   * scala> import cats.syntax.functor._
+   *
+   * scala> (5: Id[Int]).map(i => (i, i)).unzip == ((5, 5))
+   * res0: Boolean = true
+   * }}}
+   *
+   */
+  def unzip[X, Y](implicit F: Functor[F], ev: A <~< (X, Y)): (F[X], F[Y]) = F.unzip(F.map(fa)(ev))
```

Comment:
I don't think we shoud use `map` with `ev`. Instead, we should use substitution to avoid any chance we do any action. If instead we took `ev: A Is (X, Y)`, then we can use `F.unzip(ev.substitute[F](fa))`

Reply:
@johnynek sorry, I was MIA for awhile. Let me have a look, will get back to you sometime this week.

Reply:
@johnynek I've done some testing with`Is`/`Leibinz`. I had to flip the type parameters `A` and `(X, Y)` around to make it work, i.e:
```scala
def unzip[X, Y](implicit F: Functor[F], ev: (X, Y) Is A): (F[X], F[Y]) = F.unzip(ev.flip.substitute(fa))
```

Also, because tuples are covariant, there is still a possibility of a `Tuple2` to be a subtype of another `Tuple2`, eg:
```scala
    sealed trait Animal
    final case class Cat() extends Animal
    final case class Dog() extends Animal

    val l: NonEmptyList[(Cat, Dog)] = NonEmptyList one Tuple2(Cat(), Dog())
    Functor[NonEmptyList].unzip[Animal, Animal](l) //compiles for both `Is` and `As`
    l.unzip[Animal, Animal] //does not compile for `Is`
```
This would result in slight inconsistency between the prefix and postfix versions of `unzip`.


As for `As` I've tried to replace `map` with `widen` like this:
```scala
def unzip[X, Y](implicit F: Functor[F], ev: A <~< (X, Y)): (F[X], F[Y]) = F.unzip(F.widen[A, (X, Y)](fa))
```
but it would not fit the type bound. Otherwise, calling `asInstanceOf` directly would work:
```scala
def unzip[X, Y](implicit F: Functor[F], ev: A <~< (X, Y)): (F[X], F[Y]) = F.unzip(fa.asInstanceOf[F[(X, Y)]])
```
or we could implement another `widen` that requires an evidence of `As` instead of a type bound constraint:
```scala
  def widenAs[A, B: A <~< *](fa: F[A]): F[B] = fa.asInstanceOf[F[B]]
  def widen[A, B >: A](fa: F[A]): F[B] = widenAs(fa)
```

Reply:
what about this:

```scala
final class FunctorTuple2Ops[F[_], A, B](fa: F[(A, B)]) extends AnyVal {
  def _1F(implicit F: Functor[F]): F[A] = F.map(fa)(_._1)
  def _2F(implicit F: Functor[F]): F[B] = F.map(fa)(_._2)
  def unzip(implicit F: Functor[F]): (F[A], F[B]) = F.unzip(fa)
}
```
or something... sidestep using the evidence at all and only put the enrichment on Tuple2 types?

Reply:
actually I see @travisbrown suggested this originally before I derailed things: https://github.com/typelevel/cats/pull/3318#issuecomment-597524402

Reply:
@johnynek @travisbrown I've made the change to `FunctorTuple2Ops`. I think some of my comments on that might have gone to the wrong PR when I accidentally closed and reopened this PR. The original intention was to fix `unzip` directly in the `Functor` typeclass but either should be fine now since a method extension would be required either way. We might still need type evidence for efficient`unzip` for specific types.